src/app/ClusterObjectUtils.h

```diff
@@ -0,0 +1,312 @@
+/*
+ *
+ *    Copyright (c) 2020-2021 Project CHIP Authors
+ *    All rights reserved.
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+/**
+ *    @file
+ *      This file defines Base class for a CHIP IM Command
+ *
+ */
+
+#pragma once
+
+#include "core/CHIPTLVTypes.h"
+#include <app/CommandPathParams.h>
+#include <app/InteractionModelDelegate.h>
+#include <app/MessageDef/CommandDataElement.h>
+#include <app/MessageDef/CommandList.h>
+#include <app/MessageDef/InvokeCommand.h>
+#include <core/CHIPCore.h>
+#include <iterator>
+#include <messaging/ExchangeContext.h>
+#include <messaging/ExchangeMgr.h>
+#include <messaging/Flags.h>
+#include <protocols/Protocols.h>
+#include <support/BitFlags.h>
+#include <support/CodeUtils.h>
+#include <support/DLLUtil.h>
+#include <support/logging/CHIPLogging.h>
+#include <system/SystemPacketBuffer.h>
+#include <system/TLVPacketBufferBackingStore.h>
+#include <core/Optional.h>
+
+namespace chip {
+namespace app {
+namespace ClusterObjectUtils {
```

Comment:
use 'encoding' or similar instead of 'utils' in the namespace name

Reply:
See previous comment.

Suggestion: `ClusterObjectSerialization`?

Reply:
Do we expect to have more things that Serialization for cluster objects? I am wondering if we ever will have a chance of smurf naming like `ClusterObjectSerialization` and `ClusterObjectStructures` or similar. In  thatt case I would prefer 2 levels: `ClusterObject::Serialization`.

If we do not expect other 'ClusterObject' things, ClusterObjectSerialization seems fine.